src/SDK/Common/Retry/ExponentialWithJitterRetryPolicy.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Common\Retry;
+
+use InvalidArgumentException;
+use OpenTelemetry\SDK\Metrics\Exceptions\RetryableExportException;
+use OpenTelemetry\SDK\Trace\SpanExporterInterface;
+
+class ExponentialWithJitterRetryPolicy implements RetryPolicyInterface
+{
+    private $maxAttempts;
+    private $initial_backoff;
+    private $maxBackoff;
+    private $backoff_multiplier;
+    private $jitter;
+    public $retryableStatusCodes;
+
+    private function __construct(
+        int $max_attempts = self::DEFAULT_MAX_ATTEMPTS,
+        float $initial_backoff = self::DEFAULT_INITIAL_BACKOFF,
+        int $max_backoff = self::DEFAULT_MAX_BACKOFF,
+        float $backoff_multiplier = self::DEFAULT_BACKOFF_MULTIPLIER,
+        float $jitter = self::DEFAULT_JITTER,
+        array $retryableStatusCodes = null
+    ) {
+        $this->setMaxAttempts($max_attempts);
+        $this->setInitialBackoff($initial_backoff);
+        $this->setMaxBackoff($max_backoff);
+        $this->setBackoffMultipler($backoff_multiplier);
+        $this->setJitter($jitter);
+        $this->retryableStatusCodes = $retryableStatusCodes;
+    }
+
+    public function shouldRetry(
+        int $attempt,
+        int $status,
+        ?RetryableExportException $exception = null
+    ): bool {
+        if ($attempt >= $this->maxAttempts && null !== $exception) {
+            throw $exception;
+        }
+
+        return $attempt < $this->maxAttempts &&
+            $status == SpanExporterInterface::STATUS_FAILED_RETRYABLE;
+    }
+
+    public function getDelay(int $attempt): int
+    {
+        // Dont wait for the first attempt
+        if ($attempt == 0) {
+            return 0;
+        }
+        // Initial exponential backoff in milli seconds
+        $delay = ($this->backoff_multiplier ** $attempt) * $this->initial_backoff * 1000;
+        // Adding jitter to exponential backoff
+        if ($this->jitter > 0) {
+            $randomness = (int) ($delay * $this->jitter);
+            $delay = $delay + random_int(-$randomness, +$randomness);
+        }
+
+        return min((int) $delay, $this->maxBackoff * 1000);
+    }
+
+    public static function getDefault(): ExponentialWithJitterRetryPolicy
+    {
+        return new ExponentialWithJitterRetryPolicy();
+    }
+
+    public function getMaxAttempts(): int
+    {
+        return $this->maxAttempts;
+    }
+
+    public function setMaxAttempts(
+        int $max_attempts = self::DEFAULT_MAX_ATTEMPTS
+    ) {
+        $this->checkArgument(
+            $max_attempts > 0,
+            sprintf('Max attempts value must be > 0 and of int type. "%s" is given.', $max_attempts)
```

Comment:
it can't not be `int` because of the type-hint on the parameter

Reply:
I have put this check for negative value actually. Not for double.